### PR TITLE
chore: Adding fuzz tests for `filter`

### PR DESCRIPTION
**Pull request by yhakbar**
## Description

Adds fuzz tests for the `filter` package with a workflow that runs on push to fuzz everything we fuzz in the repo with a 30s budget for each.

As we add more fuzz tests, we might want to drop down that budget so that we can save some time.

## TODOs

Read the [Gruntwork contribution guidelines](https://gruntwork.notion.site/Gruntwork-Coding-Methodology-02fdcd6e4b004e818553684760bf691e).

- [ ] I authored this code entirely myself
- [x] I am submitting code based on open source software (e.g. MIT, MPL-2.0, Apache)]
- [x] I am adding or upgrading a dependency or adapted code and confirm it has a compatible open source license
- [x] Update the docs.
- [x] Run the relevant tests successfully, including pre-commit checks.
- [x] Include release notes. If this PR is backward incompatible, include a migration guide.

## Release Notes (draft)

Added / Removed / Updated [X].

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive fuzz tests to validate the filter component's robustness against diverse inputs, edge cases, and unexpected conditions.

* **Chores**
  * Enhanced the CI/CD pipeline with automated fuzz testing workflow to continuously validate functionality and catch potential issues early during development.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: internal/filter/fuzz_test.go [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+
+		// Tokenize until EOF - should never hang or panic
+		// Set a reasonable limit to prevent infinite loops in case of bugs
+		const maxTokens = 10000
+		for range maxTokens {
+			tok := lexer.NextToken()
+			if tok.Type == filter.EOF {
+				break
+			}
+		}
```
Comment (denis256) [inline internal/filter/fuzz_test.go:178]: Should we handle the maxTokens limit as an error?

```
        for range maxTokens {
            tok := lexer.NextToken()
            if tok.Type == filter.EOF {
                return
            }
        }
        t.Fatal("Lexer max token limit")

```

Comment (yhakbar) [reply internal/filter/fuzz_test.go:178]: I think we can, but isn't there a possibility that the input from the fuzzer will actually have 1000 tokens in it? I guess we'll need an upper bound error that we can throw to gracefully error on in the lexer for the bounds check to make sense here, right?
